src/App.jsx

```diff
@@ -14,45 +18,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICAL = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+
+function getGoodsPrepared(goods, sortField, reverseSort) {
+  const goodsPrepared = [...goods];
+
+  if (sortField) {
+    goodsPrepared.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICAL:
+          return a.localeCompare(b);
+
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseSort) {
+    goodsPrepared.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return goodsPrepared;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseSort, setReverseSort] = useState(false);
+  const sortedGoods = getGoodsPrepared(goodsFromServer, sortField, reverseSort);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <FancyButton
+          label="Sort alphabetically"
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICAL)}
+          isActive={sortField !== SORT_FIELD_ALPHABETICAL}
+          colorClass="is-info"
+        />
+
+        <FancyButton
+          label="Sort by length"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          isActive={sortField !== SORT_FIELD_LENGTH}
+          colorClass="is-success"
+        />
+
+        <FancyButton
+          label="Reverse"
+          onClick={() => setReverseSort(!reverseSort)}
+          isActive={!reverseSort}
+          colorClass="is-warning"
+        />
+
+        {sortField || reverseSort ? (
+          <FancyButton
+            label="Reset"
+            onClick={() => {
+              setSortField('');
+              setReverseSort(false);
+            }}
+            colorClass="is-danger is-light"
+          />
+        ) : null}
```

Comment:
As in other cases you return null, it's better just to use the `and` operator - `&&`.
Also, you can create a flag variable for `sortField || reverseSort` and it will be clearer and easier to understand.